core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -145,6 +145,13 @@ private[spark] class TaskSchedulerImpl(
   // continue to run even after being asked to decommission, but they will eventually exit.
   val executorsPendingDecommission = new HashMap[String, ExecutorDecommissionState]
 
+  // Keep removed executors due to decommission, so getExecutorDecommissionState
+  // still return correct value even after executor is lost
+  val removedExecutorsDueToDecom = new mutable.LinkedHashMap[String, ExecutorDecommissionState]
```

Comment:
Instead of manually enforcing this manually, why not use CacheBuilder ?
Something like:
`CacheBuilder.newBuilder().maximumSize(10000).expireAfterAccess(1, TimeUnit.MINUTES).build()`

(Use suitable constants above)

Reply:
Also, rename variable to `executorsRemovedByDecom` or some such.